### ISSUE TITLE
Oni frezes after performing global search

### ISSUE THREAD

**Issue by 6ruce** [onivim/oni#1973]
Config: "editor.fullScreenOnStart": true
OS: Windows 10 Home
Oni version: 0.3.1

PC: 
* RAM: 8GB
* CPU: Intel i5-6300 HQ @ 2.3GHz
* Storage: TOSHIBA MQ02ABD100H (SSHD, SSD Hybrid)

When I'm trying to perform global search (Ctrl-F) it stars working really hard and in a couple seconds whole UI frezes with blank main area and only header menu visible. After this whole PC is freazed and can't do anything except rebooting it. Looks like this can be related to the number of files in the working directory.

**Comment by badosu** [comment]
@6ruce At lot has happened since you started this issue, can you check the latest release (v0.3.4) and see if it still happens?

**Comment by karyon** [comment]
yes, it does. for me, this is reliably reproducable by opening my project folder (a medium-sized django project, but with some dependencies and some generated directories in the same folder, but these are gitignored, 13k files in total), then doing ctrl-shift-f, then slowly typing into the search field. the typing is really janky, oni will take up several gigabytes of RAM, and then the window goes blank.

**Comment by TalAmuyal** [comment]
It might be solved by https://github.com/onivim/oni/pull/2315.
@6ruce , @karyon, care to test?
Also, you can bind `"quickOpen.searchFileByContent"` to search with the QuickOpen's GUI (should be the same search/performance results though).

**Comment by iwikal** [comment]
I have this same issue. I bound `quickOpen.searchFileByContent` and when I try to use it in a decently sized folder, the whole editor grinds to a halt. Using ripgrep on its own, I get the results immediately and without issue. Does the previous query get killed when you make a new one? If not, I nominate the as-you-type searching as a possible culprit. Ripgrepping a single letter produces a lot of results.

**Comment by TalAmuyal** [comment]
By reading the code, it should kill the query and re-run the new one but it might be busted.
Anyone want to have a try at debugging it? (https://github.com/onivim/oni/blob/master/extensions/oni-plugin-quickopen/src/QuickOpen.ts#L132)
